src/files.cpp

```diff
@@ -110,4 +113,386 @@ uint8_t gzread8(gzFile f)
 	uint8_t value;
 	gzread(f, &value, 1);
 	return value;
-}
\ No newline at end of file
+}
+
+#ifndef __APPLE__
```

Comment:
As indicated by the Linux build errors, this `#ifndef __APPLE__` block and its contents should be redundant. It should be safe to remove this block, and doing so should fix the Linux builds.